### ISSUE TITLE
maestral.errors.NotFoundError when listing revisions of deleted file

### ISSUE THREAD

**Issue by SamSchott** [SamSchott/maestral#198]
## Error in Maestral

**maestral.errors.NotFoundError** in **.local/lib/python3.8/site-packages/maestral/client.py:152**
Could not list file revisions. There is nothing at the given path.

[View on Bugsnag](https://app.bugsnag.com/university-of-cambridge/maestral/errors/5f80a89e0a231500175ea5fc?event_id=5f81cd86006151f0eeab0000&i=gh&m=ci)

## Stacktrace

    .local/lib/python3.8/site-packages/maestral/client.py:152 - wrapper
    .local/lib/python3.8/site-packages/maestral/sync.py:521 - from_dbx_metadata
    .local/lib/python3.8/site-packages/maestral/sync.py:2571 - <listcomp>
    .local/lib/python3.8/site-packages/maestral/sync.py:2571 - list_remote_changes
    .local/lib/python3.8/site-packages/maestral/sync.py:3462 - startup_worker

[View full stacktrace](https://app.bugsnag.com/university-of-cambridge/maestral/errors/5f80a89e0a231500175ea5fc?event_id=5f81cd86006151f0eeab0000&i=gh&m=ci)

*Created by Sam Schott via Bugsnag*

**Comment by SamSchott** [comment]
When Maestral is notified of a remote deletion, the `DeletedMetadata` does not contain information regarding the type of the deleted item (file or folder). To query this information for our log of sync events, Maestral proceeds to list the old revisions of items at the given path, showing if the last item was a file or folder:

This query can apparently fail with a `NotFoundError` which is currently not handled and crashes the sync threads. This should not happen because if we are notified of a deletion, *something* must have been at the given path previously.

Still, it would be best to catch this error for now and file an issue with Dropbox.

**Comment by coxackie** [comment]
Latest version 1.2.1, also getting
```
Could not list file revisions
There is nothing at the given path.
```
all of a sudden (since half hour ago), and status is `Could not list file revisions`. Problem persists when I restart the daemon. Is there anything I can do to make the daemon start again?

**Comment by SamSchott** [comment]
Yes, the referenced commits do fix the issue but they are not yet included in v1.2.1. Are you using the app bundle or the Python version? If it's the latter you can check out the development branch from GitHub. It's already very stable and heading towards the next release.

**Comment by coxackie** [comment]
OK, thanks. Using the python version, and an index rebuild fixed the issue. I will just wait for the new stable.

**Comment by SamSchott** [comment]
Alright, that is safer ;)

**Comment by SamSchott** [comment]
Fixed in v1.2.2.
